src/FluentNHibernate/MappingModel/Collections/LayeredColumns.cs

```diff
@@ -17,9 +17,8 @@ public IEnumerable<ColumnMapping> Columns
                     yield break;
 
                 var maxLayer = layeredValues.Keys.Max();
-                var values = ((HashSet<ColumnMapping>)layeredValues[maxLayer]);
-
-                foreach (var value in values)
+                var values = ((Dictionary<int,ColumnMapping>)layeredValues[maxLayer]);

```

Comment:
In reference to below comment, the usage would become:
`var values = ((Dictionary<ColumnMapping, ColumnMapping>) layeredValues[maxLayer]).Keys`
`foreach (var value in values) ...`